### ISSUE TITLE
Feature wish: sub-pixel font hinting / anti-aliasing

### ISSUE THREAD

**Issue by bluenote10** [kovidgoyal/kitty#7441]
First of all: A big thank you for writing and maintaining this project; this is great!

**Is your feature request related to a problem? Please describe.**

It would be great to have sub-pixel hinting for maximizing the readability of fonts (especially at small font sizes).

Comparing e.g. `kitty` to `gnome-terminal` my immediate impression was that the fonts in `kitty` look worse than in `gnome-terminal`. Comparing the two with similar theme settings:

In `kitty` glyph rendering looks like this (magnified):

![image](https://github.com/kovidgoyal/kitty/assets/3620703/d6eaa8e6-5f3d-49b3-b47a-142e60ffd655)

Whereas in `gnome-terminal` it is rendered like this:

![image](https://github.com/kovidgoyal/kitty/assets/3620703/16364cf9-d48e-4711-a5d1-3bd75bf732ca)

It is hard to describe the difference how it actually looks on the screen, but for me it makes a pretty huge difference in terms of legibility.

**Describe the solution you'd like**

Basically "just" the common sub-pixel font hinting, but I've no idea how hard it would be to accomplish that, given that `kitty` renders fonts on the GPU.

**Describe alternatives you've considered**

Squinting my eyes a bit more :wink:

**Additional context**

Already included above.

**Comment by kovidgoyal** [comment]
dup of #214
